cm2350/gdbdefs/e200z759n3.py

```diff
@@ -0,0 +1,423 @@
+import collections
+
+
+reg_fmt = collections.OrderedDict([
+    ('r0', (32, 0)),
+    ('r1', (32, 1)),
+    ('r2', (32, 2)),
+    ('r3', (32, 3)),
+    ('r4', (32, 4)),
```

Comment:
does the data in this file match up with our emulator?
why not generate it from the register context?

Reply:
this file almost matches our register context, but not quite. Because the register groups are used by the viv gui I couldn't figure out how best to define GDB remote-server specific register groups. The alternative was trying to figure out a way to just manually specify the exact register names/widths/groups manually and I realized the way to do that was to just define the register format manually.

The biggest issue is the register width. GDB expects all of the registers to be 32 bits but they aren't defined that way in the PPC register context because we defined them based on how the PPC documentation defines the registers.

And it felt like sorting that out would take more time to figure out than I had to get this done.